padertorch/data/segment.py

```diff
@@ -0,0 +1,413 @@
+    The keys `segment_start` and `segment_stop` are added for each output
+    dictionary.
+
+    If an utterance is shorter than `length`, a
+    `lazy_dataset.FilterException` is raised.
+
+    Examples (For more examples see tests/test_data/test_segmenter:
+
+        >>> segmenter = Segmenter(length=32000, include_keys=('x', 'y'),\
+                                   shift=16000)
+        >>> ex = {'x': np.arange(65000), 'y': np.arange(65000),\
+                    'num_samples': 65000, 'gender': 'm'}
+        >>> segmented = segmenter(ex)
+        >>> type(segmented)
+        <class 'list'>
+        >>> for entry in segmented:
+        ...     print(entry['x'][[0, -1]])
+        [    0 31999]
+        [16000 47999]
+        [32000 63999]
+
+        segmenting can be disabled by setting `length=-1` or `keys=None`
+        >>> Segmenter(length=-1, include_keys=('x', 'y'))(ex)[0] == ex
+        True
+
+        Check the corner cases.
+        >>> ex = {'x': np.arange(64000), 'y': np.arange(64000)}
+        >>> for entry in segmenter(ex):
+        ...     print(entry['x'][[0, -1]])
+        [    0 31999]
+        [16000 47999]
+        [32000 63999]
+        >>> ex = {'x': np.arange(63999), 'y': np.arange(63999)}
+        >>> for entry in segmenter(ex):
+        ...     print(entry['x'][[0, -1]])
+        [    0 31999]
+        [16000 47999]
+
+    Args:
+        length: The length of the segments in samples. If set to `-1`,
+            the original example is returned in a list of length one.
+        shift: shift between segments, defaults to length
+        include_keys: The keys in the passed example dict to segment. They all
+            must have the same size along their specified `axis`, if keys is
+            None the segmentation is applied to all `numpy.arrays`.
+            If a key points to a dictionary the segmentation is applied to all
+            values of this dictionary
+        exclude_keys: This option allows to specify specific keys not to
+            segment. This might be usefull if `include_keys` is `None` or
+            one of the included keys points to a dictionary.
+        copy_keys: If `None` all not values which are not segmented are
+            copied. Otherwise only the specified keys are added to the new
+            dictionary with the segmented signals.
+        axis: axis over which to segment
+        anchor_mode: anchor mode used in `get_anchor` to calculate the anchor
+            from which the segmentation boundaries are calculated.
+        rng: random number generator (`numpy.random`)
+    """
+
+    def __init__(self, length: int = -1,shift: int = None,
+                 include_keys: Union[str, list, tuple] = None,
+                 exclude_keys: Union[str, list, tuple] = None,
+                 copy_keys: Union[str, list, tuple] = None,
+                 axis: Union[int, list, tuple, dict] = -1,
+                 anchor_mode: str = 'left', rng=np.random):
```

Comment:
For the rng, does it make sense to allow a mode where the rng is seeded from the example to have reproducibility?

Reply:
sure, I can do that. However, I though the user would specify the rng in that case.

Reply:
The user can't specify the rng when the `Segmenter` is mapped to a dataset. Then, the `Segmenter` uses the same rng for all examples

Reply:
that is True, perhaps this option should be specified during call?